nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/GuavaCacheMetric.java

```diff
@@ -0,0 +1,73 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc
+ */
+package org.nuxeo.ecm.core.storage.dbs;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.MetricSet;
+import com.google.common.cache.Cache;
+
+/**
+ * Wrapper used to wrap the Guava cache's statistics into Gauges in order to report them via Codahale Metrics.
+ *
+ * @since 8.10
+ */
+public class GuavaCacheMetric implements MetricSet {
+
+    private Map<String, Metric> metrics = new HashMap<>();
+
+    private GuavaCacheMetric() {
+    }
+
+    @Override
+    public Map<String, Metric> getMetrics() {
+        return metrics;
+    }
+
+    private <T> void putMetrics(String gaugeName, Gauge<T> gauge, String name, String... names) {
+        List<String> namesList = new ArrayList<>(Arrays.asList(names));
+        namesList.add(gaugeName);
+        metrics.put(MetricRegistry.name(name, namesList.toArray(new String[namesList.size()])), gauge);
+    }
+
+    public static MetricSet of(Cache cache, String name, String... names) {
+        GuavaCacheMetric metrics = new GuavaCacheMetric();
+        metrics.putMetrics("averageLoadPenalty", () -> cache.stats().averageLoadPenalty(), name, names);
+        metrics.putMetrics("evictionCount", () -> cache.stats().evictionCount(), name, names);
+        metrics.putMetrics("hitCount", () -> cache.stats().hitCount(), name, names);
+        metrics.putMetrics("hitRate", () -> cache.stats().hitRate(), name, names);
+        metrics.putMetrics("loadCount", () -> cache.stats().loadCount(), name, names);
+        metrics.putMetrics("loadExceptionCount", () -> cache.stats().loadExceptionCount(), name, names);
+        metrics.putMetrics("loadExceptionRate", () -> cache.stats().loadExceptionRate(), name, names);
+        metrics.putMetrics("loadSuccessCount", () -> cache.stats().loadSuccessCount(), name, names);
+        metrics.putMetrics("missCount", () -> cache.stats().missCount(), name, names);
+        metrics.putMetrics("missRate", () -> cache.stats().missRate(), name, names);
+        metrics.putMetrics("requestCount", () -> cache.stats().requestCount(), name, names);
+        metrics.putMetrics("totalLoadTime", () -> cache.stats().totalLoadTime(), name, names);
+        return metrics;
```

Comment:
here it is better to choose a dot notation for metrics because it easier to create aggregation rules on collector:
hitCount -> hit.count
hitRate -> hit.rate
averageLoadPenalty -> load.penalty.mean
load.count, load.exception.count ...